### ISSUE TITLE
Performance regression when updating from 3002 to 3029+

### ISSUE THREAD

**Issue by johnfriedrich** [SpongePowered/SpongeForge#2155]
- SpongeForge version: 3029
- Forge version: 2655
- Java version: 1.8.152

3002 timings:
https://timings.aikar.co/?id=2e1d6dc5c1d2441ba4d81a17cdfc631e
3029
https://timings.aikar.co/?id=2ea78b4029334d1c85634416ec283135
https://timings.aikar.co/?id=931c9552e58349df941608fc870e4f82

As you can see, redstone torches are the bad guys. I think sponges updates the block state on these torches every tick, even if the state (on/off) doesn't change. (My guess) Otherwise, I cannot explain this.

Timings were made on the same day, almost the same people online, doing the same stuff.

I tried 3065  once with the same bad result. Didn't had the possibility to check newer builds

**Comment by phit** [comment]
reproduce it with a clean world, and or provide steps to reproduce this.
> Timings were made on the same day, almost the same people online, doing the same stuff.

this doesn't mean anything really

**Comment by Rasgnarok** [comment]
That comment is meant to show it was tested throughout two versions without anything else changing. It doesn't replace a fresh install, but it is not useless.

**Comment by phit** [comment]
> That comment is meant to show it was tested throughout two versions without anything else changing

>Uptime | 0.44hr
>Version | 1.12.2-2611-7.1.0-BETA-3002

>Uptime | 12.12hr
>Version | 1.12.2-2655-7.1.0-BETA-3029

k..

**Comment by johnfriedrich** [comment]
3002:
https://timings.aikar.co/?id=65241d99a9e04aedae22718056341689

Redstone lags the server. Runtime doesn't matter in that case.

**Comment by Rasgnarok** [comment]
Regardless of uptime, this isn't expected behaviour, nowhere near it in fact. If you need more detailed testing or description, please, let us know.

**Comment by bloodmc** [comment]
@i9hdkill What was the first SF build post 3002 that started with issues? 3029? Also, please provide a warmroast trace when the server lags.

**Comment by bloodmc** [comment]
@i9hdkill Please turn off `block-tick-capturing` and retest. It might even be best to test with a CLEAN sponge config.

**Comment by bloodmc** [comment]
Actually `block-tick-capturing` seems to have been broken for quite some time so its definitely not that.

**Comment by bloodmc** [comment]
I really don't see anything between 3002 and 3029 that would cause redstone to blow up the server. It may be a Forge change that is causing the performance regression. Have you compared Forge 2611 and Forge 2655+ for performance differences?

**Comment by bloodmc** [comment]
I cannot reproduce any redstone issues using latest build. I tested on the same redstone map I've used for years

https://www.planetminecraft.com/project/some-redstone-creations/

Logging into the world , the server sits at around 4ms. If I start to turn devices on then it starts to rise above 10ms. All torches are also working as they should.

**Comment by bloodmc** [comment]
Closing as there has been no response in over a week.

**Comment by johnfriedrich** [comment]
Sponge: 3002
Forge 2611
Warmroast:
part1: http://prntscr.com/jmjgvm
part2: http://prntscr.com/jmjhia
Timings: https://timings.aikar.co/?id=0e94c8fcf2a94911bb1fbdb5e98c0fa9

Yes, seems to be also on older builds, like you said

**Comment by johnfriedrich** [comment]
Found it. It's an massive autofarm that worked without lags in sponge 1.10.
But the issue only arised some weeks ago, as the guy who build it got active again at that time. Around that time, we updated from 3002 to later sponge. Thats why we thought it's the sponge version.

Still, it works without that much lag on 1.10.2 sponge.

Schematic: https://www.dropbox.com/s/geuloyjqu7ead4b/TheWhiteBlader_farm.schematic?dl=0
